pkg/apis/certificates/types.go

```diff
@@ -43,6 +43,10 @@ type CertificateSigningRequestSpec struct {
 	// Base64-encoded PKCS#10 CSR data
 	Request []byte `json:"request"`
 
+	// signingProfile specifies a requested signing profile for a CA to use
+	// when issuing the certificate.
+	// SigningProfile string `json:"signingProfile,omitempty"`
```

Comment:
Thinking about how the CSR API will be used, clients will need specific uses (node needs client cert for API contact, server cert for serving its API). They might have a preference for expiration, or might not. Making those clients look up signing profiles (and choose between them if multiple satisfy their needs) seems unnecessary, and loses information, since the approver doesn't know which of the uses in the referenced profile the requester actually needed. Taking our first and second use cases (kubelet client CSR and kubelet serving cert CSR), if the requester can specify required uses directly, it's trivial to submit a request. If it has to reference a profile instead, the logic becomes much much harder